abdiff/core/run_ab_transforms.py

```diff
@@ -218,24 +221,27 @@ def run_docker_container(
         },
         volumes=[f"{os.path.abspath(transformed_directory)}:/tmp"],
     )
-    logger.info(
-        f"Container '{container.id}' (Docker image: {docker_image}) "
-        f"RUNNING transform for '{source}' input_file: {input_file}."
-    )
+    logger.info(f"Transmogrifier container ({container.short_id}) STARTED: {input_file}")
 
     exception = None
     try:
         start_time = perf_counter()
         while True:
             time.sleep(0.5)
             container.reload()
+            elapsed_time = time.perf_counter() - start_time
             if container.status == "exited":
-                logger.info(f"Container {container.id} exited.")
+                logger.info(
+                    f"Transmogrifier container ({container.short_id}) EXITED, "
+                    f"elapsed {timedelta(seconds=elapsed_time)}: {input_file}"
+                )
+                write_log_file(run_directory, input_file, container)
                 break
 
             if time.perf_counter() - start_time > timeout:
                 logger.error(
-                    f"Container {container.id} timed out after {timeout} seconds"
+                    f"Transmogrifier container ({container.short_id}) TIMED OUT, "
+                    f"elapsed {timedelta(seconds=elapsed_time)}: {input_file}"
                 )
                 container.stop()
```

Comment:
After we stop the container which we have determined exceeds our timeout, I wonder if we want to write it's logs still?  Instead of "Transmogrifier container ({container.short_id}) EXITED" we could write something to the header like "Transmogrifier container ({container.short_id}) TIMED OUT".

Reply:
Updated so that the app will write any available logs for containers in the event of a timeout and append a message at the end to indicate timeout. See example:

```text
docker_image: transmogrifier-abdiff-3f77c7c:latest | source: libguides | input_file: s3://timdex-extract-prod-300442551476/libguides/libguides-2024-11-13-daily-extracted-records-to-index.xml
container: 0d70653ef4f7f938b5c3313bad141051d7b15a113d836a00011ce3abe055b559
2024-11-15 17:10:40,433 INFO transmogrifier.cli.main(): Logger 'root' configured with level=INFO
2024-11-15 17:10:40,433 INFO transmogrifier.cli.main(): No Sentry DSN found, exceptions will not be sent to Sentry
2024-11-15 17:10:40,433 INFO transmogrifier.cli.main(): Running transform for source libguides
2024-11-15 17:10:40,441 INFO botocore.credentials.load(): Found credentials in environment variables.
2024-11-15 17:10:40,505 INFO botocore.configprovider.provide(): Found endpoint for s3 via: environment_global.
2024-11-15 17:10:40,548 INFO transmogrifier.cli.main(): Completed transform, total records processed: 10, transformed records: 10, skipped records: 0, deleted records: 0
2024-11-15 17:10:40,548 INFO transmogrifier.cli.main(): Total time to complete transform: 0:00:00.114957
Transmogrifier container (0d70653ef4f7) TIMED OUT
```